plotting_functions.py

```diff
@@ -0,0 +1,1653 @@
+    hazard_prob_of_exceedance = calculators.rate_to_prob(hazard_rate_array, 1.0)
+
+    resulting_hazard_curves = (
+        plotting_helpers.load_aggregate_stats_for_all_logic_trees_in_directory(
+            results_directory.parent
+        )
+    )
+
+    agg_stats_df = resulting_hazard_curves.data_df
+
+    ## Get the needed mean and standard deviation values
+    mean = agg_stats_df[
+        (agg_stats_df["agg"] == "mean")
+        & (agg_stats_df["vs30"] == vs30)
+        & (agg_stats_df["imt"] == im)
+        & (agg_stats_df["hazard_model_id"] == results_directory.name)
+        & (agg_stats_df["nloc_001"] == locations_nloc_dict[location_short_name])
+    ]["values"].values[0]
+
+    std_ln = agg_stats_df[
+        (agg_stats_df["agg"] == "std_ln")
+        & (agg_stats_df["vs30"] == vs30)
+        & (agg_stats_df["imt"] == im)
+        & (agg_stats_df["hazard_model_id"] == results_directory.name)
+        & (agg_stats_df["nloc_001"] == locations_nloc_dict[location_short_name])
+    ]["values"].values[0]
+
+    plot_ylims = (1e-5, 1)
+    _, axes = plt.subplots(1, 2, figsize=(8, 5))
+
+    for realization_index in range(len(hazard_prob_of_exceedance)):
+        gmcm_name_with_branch = realization_names[
+            realization_index
+        ].ground_motion_characterization_models_id
+        gmcm_name = gmcm_name_with_branch.split("(")[0]
+
+        if gmcm_name_with_branch in plot_label_lookup.keys():
+            plot_label = plot_label_lookup[gmcm_name_with_branch]
+        else:
+            plot_label = gmcm_name_with_branch
+
+        axes[0].loglog(
+            nshm_im_levels,
+            hazard_prob_of_exceedance[realization_index],
+            label=plot_label,
+            marker="o",
+            color=plot_colors[realization_index],
+            linestyle=plot_linestyles[realization_index],
+        )
+
+    # Customize the second subplot
+    # noinspection PyUnboundLocalVariable
+    if gmcm_name in gmcm_name_formatting_lookup.keys():
+        gmcm_name_label = gmcm_name_formatting_lookup[gmcm_name]
+    else:
+        gmcm_name_label = gmcm_name
+
+    axes[1].semilogy(
+        std_ln, mean, marker="o", linestyle="-", color="tab:blue", label=gmcm_name_label
+    )
+
+    axes[0].legend(loc="lower left")
+    axes[1].legend(loc="lower left")
+
+    for ax_index in range(len(axes)):
+        axes[ax_index].grid(
+            which="major", linestyle="--", linewidth="0.5", color="black", alpha=0.5
+        )
+
+        axes[ax_index].set_ylim(plot_ylims)
+    axes[0].set_xlim(9e-5, 5)
+    axes[1].set_xlim(-0.01, 0.68)
+
+    ### The annotations for explanation
+    annotation_ims = [1e-2, 1e-1, 1e0]
+
+    ### latex formatted strings corresponding to the annotation_ims.
+    ### These could be generated automatically from the annotation_ims
+    ### but they are just hard coded for simplicity.
+    manually_matched_latex_strings = [r"10$^{-2}$", r"10$^{-1}$", r"10$^{0}$"]
```

Comment:
```suggestion
    manually_matched_latex_strings = ["$10^{-2}$", "$10^{-1}$","$10^{0}$"]
```

Reply:
these strings do not need to be raw strings because they don to contain any escape characters.